doc/mgr/orchestrator_cli.rst

```diff
@@ -8,54 +8,229 @@ Orchestrator CLI
 
+As the orchestrator CLI unifies different external orchestrators, a common nomenclature
+for the orchrstrator module is needed. 
+
++--------------------------------------+--------------------------------------+
+| host                                 | hostname (not DNS name) on the       |
+|                                      | physical host. Not the podname,      |
+|                                      | container name, or hostname inside   |
+|                                      | the container.                       |
++--------------------------------------+--------------------------------------+
+| service                              | A logical service, e.g., “MDS for    |
+|                                      | fs1” or “NFS gateway(s)”. Typically  |
+|                                      | comprised of multiple service        |
+|                                      | instances on multiple hosts for HA   |
++--------------------------------------+--------------------------------------+
+| service instance                     | A single instance of a service.      |
+|                                      |  Usually a daemon, but maybe not     |
+|                                      | (e.g., might be a kernel service     |
+|                                      | like LIO or knfsd or whatever)       |
++--------------------------------------+--------------------------------------+
+| daemon                               | A running process on a host; use     |
+|                                      | “service instance” instead           |
++--------------------------------------+--------------------------------------+
+
 Configuration
 =============
 
-You can select the orchestrator module to use with the ``set backend`` command:
-
-::
+You can select the orchestrator module to use with the ``set backend`` command::
 
     ceph orchestrator set backend <module>
 
-For example, to enable the Rook orchestrator module and use it with the CLI:
-
-::
+For example, to enable the Rook orchestrator module and use it with the CLI::
 
     ceph mgr module enable orchestrator_cli
     ceph mgr module enable rook
     ceph orchestrator set backend rook
 
 
-You can then check backend is properly configured:
-
-::
+You can then check backend is properly configured::
 
     ceph orchestrator status
 
 
 Usage
 =====
 
+.. note::
+
+    Orchestrator modules may only implement a subset of the commands listed below.
+
+Status
+~~~~~~
+
+::
+
+    ceph orchestrator status
+
+Show current orchestrator mode and high-level status (whether the module able
+to talk to it)
+
+Also show any in-progress actions.
+
+
+Host Management
+~~~~~~~~~~~~~~~
+
+List hosts accociatted with the cluster::
+
+    ceph orchestrator host ls
+
+Add and remove hosts::
+
+    ceph orchestrator host add <host>
+    ceph orchestrator host rm <host>
+
+.. note::
+
+    Removing a host only succeeds, if the host is unused.
+
```

Comment:
This command (and several others) are current not implemented, even in the CLI code. "orchestrator host" is currently not recognized and you'll get back "no valid command found" if you try to use it.

I know it says that not all commands are implemented in all orchestrator backends, but should we make note of that fact here so that no one is misled into thinking this command should work?

Reply:
Yep, this should be more like "this is the future cli definition, everything is WIP"